packages/docusaurus-theme-classic/src/theme/DocSidebar/index.tsx

```diff
@@ -78,20 +85,30 @@ function DocSidebarItemCategory({
     if (justBecameActive && collapsed) {
       setCollapsed(false);
     }
-  }, [isActive, wasActive, collapsed]);
-
-  const handleItemClick = useCallback(
-    (e) => {
-      e.preventDefault();
-
-      if (!menuListHeight) {
-        handleMenuListHeight();
+    if (goTo && collapsed) {
+      for (const i in items) {
+        if (items.hasOwnProperty(i)) {
+          const childItem = items[i];
+          if (childItem.type === 'link') {
+            if (goTo.includes(childItem.label.toLowerCase())) {
```

Comment:
That works but does not really feel like a reliable  implementation.
What we want is to link to a doc by using its id, not a doc label.
The label of a doc will be  translated on i18n sites, so it's not a reliable way to say "I want this category to link to that doc"

Let me know if you don't find a way to link to doc ids, it might not be easily accessible, I can take a look.

Reply:
I've been trying to see a way to get link the doc ids but I haven't found a sound a solution any ideas on how I can tackle this?

Reply:
@slorber I tried something similar in the past and remember it being quite difficult to get the id within this context. I would love to know your approach.

Reply:
This probably has to be implemented directly in the docs plugin, so that the DocSidebar components directly receive the target link destination instead of receiving the docId and trying to get back the link destination.

Much more data is available in the docs plugin, much less is available through props (it wouldn't be optimized to pass too much props), so we try to only pass the minimum required

Reply:
the idea is also to move the complexity from the theme to the docs plugin. We want to have multiple themes and don't want to duplicate complicated theme logic in each of them.